app/controllers/edit_directory_columns_controller.rb

```diff
@@ -26,16 +24,30 @@ def update
       raise Discourse::InvalidParameters, "Must have at least one column enabled"
     end
 
+    changes = "\n"
     directory_column_params[:directory_columns].values.each do |column_data|
       existing_column = directory_columns.detect { |c| c.id == column_data[:id].to_i }
       if (
-           existing_column.enabled != column_data[:enabled] ||
+           existing_column.enabled != ActiveModel::Type::Boolean.new.cast(column_data[:enabled]) ||
              existing_column.position != column_data[:position].to_i
          )
-        existing_column.update(enabled: column_data[:enabled], position: column_data[:position])
+        changes +=
+          "#{existing_column.name} - enabled: #{existing_column.enabled} -> #{column_data[:enabled]} , position: #{existing_column.position} -> #{column_data[:position]}\n"
+        existing_column.update(
+          enabled: column_data[:enabled],
+          position: column_data[:position].to_i,
+        )
       end
     end
 
+    details =
+      if changes.empty?
+        { Detail: "Nothing was changed" }
```

Comment:
This needs localization too, but a moot point if we switch to a specific `log_` function.

Reply:
Would it make sense to skip logging when nothing changed?